### PR TITLE
Port WebGL rendering to (experimental) WebGPU

### PR DESCRIPTION
**Pull request by smolck**
Still have a good bit to go before this is done, but gonna go ahead and open the PR now. Requires a switch to `electron-nightly` and the usage of the `enable-unsafe-webgpu` flag, TBD on whether or not I merge this once it's done because of that.

Regardless of the mergeability of this, if you're curious why I want to make this switch, just look at the old WebGL code and the new WIP WebGPU code for starters; it's a lot nicer to use/work with and understand, at least for graphics code. In addition, WebGPU is a better API for modern graphics hardware, and can therefore apparently be anywhere from 2-8 times faster than an equivalent (ish?) WebGL implementation. Remains to be seen whether or not this adds to the performance of uivonim, but I have hopes it will.

In addition, since the API is easier to use and understand compared to WebGL, and with the added feature of compute shaders (and others probably), I'm more confident I'll be able to do fun effects and try new things, such as adding a bloom effect to mimic [this VSCode plugin](https://marketplace.visualstudio.com/items?itemName=RobbOwen.synthwave-vscode), perhaps allowing for playing interesting backgrounds instead of just a color like in the TUI, etc.

First goal though is to port the existing WebGL code to WebGPU, which is what this PR will hopefully accomplish. Whether or not the WebGL renderer is kept, whether the WebGPU renderer will be the default should I get it to work as good as (if not better than) the existing renderer, etc., are TBD for now.

## Related Links/Extra Info
Implementation status of WebGPU in browsers: https://github.com/gpuweb/gpuweb/wiki/Implementation-Status
WebGPU spec: https://gpuweb.github.io/gpuweb/
WGSL (eventual shading language of WebGPU) spec: https://gpuweb.github.io/gpuweb/wgsl.html
WebGPU repo: https://github.com/gpuweb/gpuweb

### CODE REVIEW

**Comment by smolck** [comment]
I may come back to this in the future, but I don't really see myself working on this anytime soon. WebGPU has a ways to go before it's as stable as WebGL(2), and I'm not really sure any benefits gained are worth the effort it would take to make this work (and the usage of electron-nightly and the unsafe webgpu flag, etc.). 

Plus, I'm planning on rewriting the renderer and corresponding parts (as it makes sense) in Rust with [luminance](https://github.com/phaazon/luminance-rs) compiled to wasm anyways (have actually already started work on re-implementing the font-atlas part of the rendering code in Rust), which only compiles to WebGL(2) for the web and not WebGPU.

TL;DR: Sticking with WebGL for the time being.
